.github/workflow_scripts/lint_check.sh

```diff
@@ -6,3 +6,5 @@ source $(dirname "$0")/env_setup.sh
 
 setup_build_env
 black --check --diff multimodal/src/autogluon/multimodal
+black --check --diff timeseries/
```

Comment:
let's please have time series lint check as a separate CI step, running in parallel with `multimodal` lint check?

Reply:
I think this option will result in a noticeable amount of redundant code in [`.github/workflows/continuous_integration.yml`](https://github.com/awslabs/autogluon/blob/master/.github/workflows/continuous_integration.yml) and may potentially hurt future extensibility:
- We will need to add `lint_check_timeseries` and `lint_check_multimodal` as a prerequisite (`needs: ...`) to many other actions. 
- If other modules decide to add linting they will again need to extend `needs` list for every existing action as well as add a separate action to the list. However, with the current solution where all linting happens in a single action people will only need to add a single line of code.
- The actual execution time of `black ...` and `isort ...` is negligible compared to setting up new environments, most of the time is spent creating a test environment - we can remove this overhead and not create new actions.

Reply:
For the first two points, we seem to have paid this price for other cases such as unit tests and tutorials I think it's OK to pay it here. Also, @yinweisu your comments would be welcome.

I also agree with the final point, but as the overall setup cost is negligible I would prefer to have it as a standalone item in the graph just for usability's sake (and to not have lint fails of some modules masking others if we keep scaling in this manner).

Reply:
@shchur Thanks for the contribution! I think the duplicate code is not too much of a concern as @canerturkmen has discussed. However, we do pay an overhead of spinning up aws batch machines and downloading image, which is not cheap compared to lint check. Maybe we can start with having lint checks all together, and we can always change that later easily if we want.

Reply:
thanks @yinweisu . resolving this then.